tests/translator/test_translator.py

```diff
@@ -29,10 +31,34 @@ def deep_sorted(value):
     if isinstance(value, dict):
         return {k: deep_sorted(v) for k, v in value.items()}
     if isinstance(value, list):
-        return sorted(deep_sorted(x) for x in value)
+        # To keep Py2 consistent with the previous state, we keep the sort the same
+        if sys.version_info[0] < 3:
+            return sorted((deep_sorted(x) for x in value))
+        else:
+            # In Py3, how lists are sorted and compared has changed. To keep this as consistent as possible with Py2,
+            # we need to provide the key kwarg
+            return sorted((deep_sorted(x) for x in value), key=cmp_to_key(py2_compare_order))
     else:
         return value
 
+def py2_compare_order(obj1, obj2):
+    """
+    This is a comparison function that mimics the results of Py2 Comparisons.
+    """
+    if isinstance(obj1, dict) and isinstance(obj2, dict):
+        obj1 = json.dumps(obj1, sort_keys=True)
+        obj2 = json.dumps(obj2, sort_keys=True)
```

Comment:
No way does this mimic the python 2 comparison order

Reply:
I think the code you're trying to emulate is:

https://github.com/python/cpython/blob/2.7/Objects/object.c#L767-L807

Reply:
@eric-wieser Rewriting the complete sort order is not the goal here. We wanted to get mimic the functionality of Py2 until we can make a deeper pass at the tests to assert not only the template but the order on which values are in lists (some lists in a template must be ordered). This is called out in the second bullet point in the description of the PR.